crates/bevy_render/src/lib.rs

```diff
@@ -66,14 +71,33 @@ use crate::{
     view::{ViewPlugin, WindowRenderPlugin},
 };
 use bevy_app::{App, AppLabel, Plugin, SubApp};
-use bevy_asset::{load_internal_asset, AssetApp, AssetServer, Handle};
+use bevy_asset::{AssetApp, AssetServer};
 use bevy_ecs::{prelude::*, schedule::ScheduleLabel, system::SystemState};
 use bevy_utils::tracing::debug;
 use std::{
     ops::{Deref, DerefMut},
     sync::{Arc, Mutex},
 };
 
+// TODO: this should be replaced by proper asset handling in the shader loader.
+/// Inline shader as an `embedded_asset` and load it permanently.
+///
+/// This works around a limitation of the shader loader not properly loading
+/// dependencies of shaders.
+#[macro_export]
+macro_rules! load_and_forget_shader {
```

Comment:
Since `AssetServer::load` is async, it is (I believe) theoretically possible that shaders loaded in this way are not fully loaded before they are needed. If shader A depends on shader B, and A is loaded while B is not, it can cause errors when the render pipelines try to use A.

There seems to be no easy way to work around this. Maybe it does not matter in practice as loading from the embedded asset source should be fast, but this at least deserves mentioning in comments.